src/Nethermind/Nethermind.Consensus/Processing/RecoverSignature.cs

```diff
@@ -114,6 +119,32 @@ public void RecoverData(Block block)
                     if (_logger.IsTrace) _logger.Trace($"Recovered {tx.SenderAddress} sender for {tx.Hash}");
                 }
             }
+
+            if (releaseSpec.IsAuthorizationListEnabled)
+            {
+                foreach (Transaction tx in block.Transactions.AsSpan())
```

Comment:
Perhaps, it would be better to run this loop in parallel as well (not sure though)

Reply:
I think the whole function could be done in more optimal way